### DISCUSSION TITLE
Support For FlowGem (Hive Leak Sensor)

### DISCUSSION THREAD

**Discussion by BEMSman** [arendst/Tasmota#15619] [Ideas]
Hello,
          Just wondered if there may be any custom firmware for the FlowGem/Hive HLS001 water leak sensor?

This device has the ESP-WROOM-02 (ESP-8266 Family), Winbond 25q40cl SPI Flash & Atmel ATSAMD20E16 ARM processor.

Attached is the front and back of the PCB
![IMG_1456](https://user-images.githubusercontent.com/75639810/168422705-3e32de73-b38b-435a-afa6-18561e192b13.jpg)
![IMG_1453](https://user-images.githubusercontent.com/75639810/168422708-8820e4e4-fc39-4de5-8049-b99715dc3c8d.jpg)
.

**Comment by barbudor** [comment]
If there's an ARM it's likely be the controller of the unit, waking up the ESP only when needed.
And yes the ARM has a proprietary firmware.

Before flashing Tasmota you need to find out the connections and analyse the protocol
